catalog/dags/maintenance/rotate_db_snapshots.py

```diff
@@ -9,38 +9,62 @@
 The DAG will automatically delete the oldest snapshots when more snaphots
 exist than it is configured to retain.
 
-Requires two variables:
+Relies on three variables:
 
-`AIRFLOW_RDS_ARN`: The ARN of the RDS DB instance that needs snapshots.
-`AIRFLOW_RDS_SNAPSHOTS_TO_RETAIN`: How many historical snapshots to retain.
+`CATALOG_RDS_DB_IDENTIFIER`: (Required) The "DBIdentifier" of the RDS DB instance.
+`CATALOG_RDS_SNAPSHOTS_TO_RETAIN`: (Required) How many historical snapshots to retain.
+`CATALOG_RDS_REGION`: (Optional) The region of the RDS DB instance. Defaults to
+`us-east-1`.
 """
 
 import logging
 from datetime import datetime
 
-import boto3
 from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.providers.amazon.aws.hooks.rds import RdsHook
 from airflow.providers.amazon.aws.operators.rds import RdsCreateDbSnapshotOperator
 from airflow.providers.amazon.aws.sensors.rds import RdsSnapshotExistenceSensor
 
+from common.constants import AWS_RDS_CONN_ID
+
 
 logger = logging.getLogger(__name__)
 
 DAG_ID = "rotate_db_snapshots"
 MAX_ACTIVE = 1
+# This cannot be pulled in the DAG itself because ``hook_params``
+# on the operators provided by the amazon provider is not templated
+RDS_REGION = Variable.get("CATALOG_RDS_REGION", "us-east-1")
```

Comment:
I wonder - I didn't need to supply a region in #2099 when I specify it in the Airflow connection parameter: `AIRFLOW_CONN_AWS_RDS=aws://<id>:<secret>@?region_name=us-east-1`

We really want to try and avoid `Variable.get`s in DAG files since they get run (unnecessarily) every time the DAGs are parsed. Are we able to remove this if we define the region using the conn ID now that we're using a separate RDS connection?

Reply:
I would assume so.